testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/events/EventMetricsProviderTest.java

```diff
@@ -0,0 +1,165 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.testsuite.events;
+
+import io.micrometer.core.instrument.Metrics;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.Test;
+import org.keycloak.common.Profile;
+import org.keycloak.events.Details;
+import org.keycloak.events.Errors;
+import org.keycloak.events.EventBuilder;
+import org.keycloak.events.EventType;
+import org.keycloak.models.RealmModel;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.testsuite.AbstractKeycloakTest;
+import org.keycloak.testsuite.arquillian.annotation.EnableFeature;
+import org.keycloak.testsuite.arquillian.containers.AbstractQuarkusDeployableContainer;
+import org.keycloak.testsuite.util.RealmBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.keycloak.testsuite.auth.page.AuthRealm.TEST;
+
+/**
+ * @author aschwart
+ */
+@EnableFeature(Profile.Feature.USER_EVENT_METRICS)
+public class EventMetricsProviderTest extends AbstractKeycloakTest {
+
+    private final static String CLIENT_ID = "CLIENT_ID";
+
+    private void enableUserEventMetrics(String tags, String events) {
+        try {
+            suiteContext.getAuthServerInfo().getArquillianContainer().getDeployableContainer().stop();
+            enableEventMetricsOptions(tags, events);
+            suiteContext.getAuthServerInfo().getArquillianContainer().getDeployableContainer().start();
+            reconnectAdminClient();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void enableEventMetricsOptions(String tags, String events) {
+        if (suiteContext.getAuthServerInfo().isQuarkus()) {
+            AbstractQuarkusDeployableContainer container = (AbstractQuarkusDeployableContainer) suiteContext.getAuthServerInfo().getArquillianContainer().getDeployableContainer();
+            List<String> args = new ArrayList<>();
+            args.add("--metrics-enabled=true");
+            args.add("--event-metrics-user-enabled=true");
+            if (tags != null) {
+                args.add("--event-metrics-user-tags=" + tags);
+            }
+            if (events != null) {
+                args.add("--event-metrics-user-events=" + events);
+            }
+            container.setAdditionalBuildArgs(args);
+        }
+        else {
+            setConfigProperty("keycloak.metrics-enabled", "true");
+            setConfigProperty("keycloak.event-metrics-user-enabled", "true");
```

Comment:
I believe this won't work without proper modifications of the [Undertow config file](https://github.com/keycloak/keycloak/blob/9a7cfb38ac7fc238d9d35f81df4c675adb5189ac/testsuite/integration-arquillian/tests/base/src/test/resources/META-INF/keycloak-server.json).

Also it's missing the optional settings for tags and events.

The question is: do we care? Undertow is no longer used in any of ours CIs, AFAIK. It's used only for dev purposes. We could simply say "this test runs only on Quarkus".

The questions